### PR TITLE
Properly escape test names when setting PYTEST_CURRENT_TEST environment variable

### LINKED ISSUE #2644: pytest 3.2 breaks pytest.mark.parametrize with null bytes
```
(tempenv-543d64822ecc) ~/.v/tempenv-543d64822ecc ❯❯❯ pip install "pytest<3.2"
Collecting pytest<3.2
  Using cached pytest-3.1.3-py2.py3-none-any.whl
Requirement already satisfied: py>=1.4.33 in ./lib/python2.7/site-packages (from pytest<3.2)
Requirement already satisfied: setuptools in ./lib/python2.7/site-packages (from pytest<3.2)
Installing collected packages: pytest
  Found existing installation: pytest 3.2.0
    Uninstalling pytest-3.2.0:
      Successfully uninstalled pytest-3.2.0
Successfully installed pytest-3.1.3
(tempenv-543d64822ecc) ~/.v/tempenv-543d64822ecc ❯❯❯ cat test.py
import pytest


@pytest.mark.parametrize("data", ["\x00"])
def test_foo(data):
    assert data
(tempenv-543d64822ecc) ~/.v/tempenv-543d64822ecc ❯❯❯ py.test  test.py
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /Users/alex_gaynor/.virtualenvs/tempenv-543d64822ecc, inifile:
collected 1 item s

test.py .

==================================================== 1 passed in 0.01 seconds ====================================================
(tempenv-543d64822ecc) ~/.v/tempenv-543d64822ecc ❯❯❯ pip install -U pytest
Collecting pytest
  Using cached pytest-3.2.0-py2.py3-none-any.whl
Requirement already up-to-date: py>=1.4.33 in ./lib/python2.7/site-packages (from pytest)
Requirement already up-to-date: setuptools in ./lib/python2.7/site-packages (from pytest)
Installing collected packages: pytest
  Found existing installation: pytest 3.1.3
    Uninstalling pytest-3.1.3:
      Successfully uninstalled pytest-3.1.3
Successfully installed pytest-3.2.0
(tempenv-543d64822ecc) ~/.v/tempenv-543d64822ecc ❯❯❯ py.test  test.py
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.13, pytest-3.2.0, py-1.4.34, pluggy-0.4.0
rootdir: /Users/alex_gaynor/.virtualenvs/tempenv-543d64822ecc, inifile:
collected 1 item s

test.py EE

============================================================= ERRORS =============================================================
_________________________________________________ ERROR at setup of test_foo[] __________________________________________________

self = {'TERM_PROGRAM_VERSION': '3.0.15', 'LOGNAME': 'alex_gaynor', 'USER': 'alex_gaynor', 'HOME': '/Users/alex_gaynor', 'PAT..._ENCODING': '0x1F5:0x0:0x0', 'PWD': '/Users/alex_gaynor/.virtualenvs/tempenv-543d64822ecc', 'COMMAND_MODE': 'unix2003'}
key = 'PYTEST_CURRENT_TEST', item = 'test.py::test_foo[\x00] (setup)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: putenv() argument 2 must be string without null bytes, not str

lib/python2.7/os.py:473: TypeError
________________________________________________ ERROR at teardown of test_foo[] ________________________________________________

self = {'TERM_PROGRAM_VERSION': '3.0.15', 'LOGNAME': 'alex_gaynor', 'USER': 'alex_gaynor', 'HOME': '/Users/alex_gaynor', 'PAT..._ENCODING': '0x1F5:0x0:0x0', 'PWD': '/Users/alex_gaynor/.virtualenvs/tempenv-543d64822ecc', 'COMMAND_MODE': 'unix2003'}
key = 'PYTEST_CURRENT_TEST', item = 'test.py::test_foo[\x00] (teardown)'

    def __setitem__(self, key, item):
>       putenv(key, item)
E       TypeError: putenv() argument 2 must be string without null bytes, not str

lib/python2.7/os.py:473: TypeError
==================================================== 2 error in 0.07 seconds =====================================================
(tempenv-543d64822ecc) ~/.v/tempenv-543d64822ecc ❯❯❯
```

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #2644

### CODE REVIEW

**Comment by The-Compiler** [comment]
I wonder what other stuff null bytes in node IDs could break though...

**Comment by The-Compiler** [comment]
I think `_ascii_escaped` does nothing here (like Ronny says, `_ascii_escaped` is probably called on all node IDs already) - however, null bytes are in ASCII :wink:

I think the proper solution for this is for `_ascii_escaped` to escape null bytes (probably as `\x00` for consistency though).

**Comment by nicoddemus** [comment]
> this needs a followup as test ids should be ascii to begin with, if they are non-ascii then we messed up somewhere else
> I think _ascii_escaped does nothing here 

Argh you guys are correct, it was a brain fart on my part: nodeids are already turned into `ascii` during parametrization. The problem here is that "null bytes" *are* ascii, so they are left unchanged. This works fine everywhere else it seems, but Python 2 (and not Python 3) doesn't like null bytes in environment variables, due to implementation details (I assume).

I would not like to touch `_ascii_escaped` to remove *null bytes* at the moment, given that it seems to work fine everywhere else so far; we might decide that it is safer for the long term to do so, but I would rather do it in `features` if we decide to do so to ensure we don't break anything now.

I changed it to only remove the null bytes from the string that will be set as environment variable, and only in Python 2. Let me know what you guys think.

**Comment by RonnyPfannschmidt** [comment]
a valid change, but it will also mean that the ids we store will not match  the real test name, in addition i suspect we cant really support ids with null bytes as they cant be properly transferred anywhere else or even entered from the cli

File: _pytest/runner.py [hunk 1]
```diff
@@ -134,7 +135,11 @@ def _update_current_test_var(item, when):
     """
     var_name = 'PYTEST_CURRENT_TEST'
     if when:
-        os.environ[var_name] = '{0} ({1})'.format(item.nodeid, when)
+        value = _ascii_escaped('{0} ({1})'.format(item.nodeid, when))
```
Comment (The-Compiler) [inline _pytest/runner.py]: So does this `_ascii_escaped` call do anything here? I think it does nothing as we discovered, and if that's true, it should probably be removed to not cause further confusion.

Comment (nicoddemus) [inline _pytest/runner.py]: Sorry, I didn't push the changes yet (I wrote my comment meaning to push right away, but got dragged by other things).

Comment (nicoddemus) [inline _pytest/runner.py]: Just pushed it

**Comment by nicoddemus** [comment]
>  but it will also mean that the ids we store will not match the real test name

What do you mean by "will not"? I mean: we already "ascii escape" node ids for some time now, so I don't see why my PR changed this in any way.

> in addition i suspect we cant really support ids with null bytes as they cant be properly transferred anywhere else or even entered from the cli

I see what you mean, but as I said I'm not comfortable doing this in this PR (or in `master` for that matter) because we might break things that are working. I'm OK with changing that in `features` (and in a separate PR) if we think that's a good idea.

**Comment by The-Compiler** [comment]
I agree with @nicoddemus, and I think it'd be a good idea to escape null bytes in node IDs in general (in `features`). I'm kind of surprised this didn't bite us earlier, considering that node IDs are used in various places.

Actually, this makes me wonder whether we should just escape all ASCII values before 0x20 (space)... Those are all unprintable, and some of them invisible (but it also contains things like tab).

**Comment by nicoddemus** [comment]
Moved this discussion over to #2649
